master/internal/experiment/bulk_action.go

```diff
@@ -455,16 +490,14 @@ func ArchiveExperiments(ctx context.Context, system *actor.System,
 		Model(&expChecks).
 		Column("e.archived").
 		Column("e.id").
-		ColumnExpr("e.state IN (?) AS state", bun.In([]string{
-			"CANCELED",
-			"COMPLETED",
-			"ERROR",
-		}))
+		ColumnExpr("e.state IN (?) AS state", bun.In(terminalExperimentStates))
 
 	if filters == nil {
 		query = query.Where("e.id IN (?)", bun.In(experimentIds))
 	} else {
-		query = queryBulkExperiments(query, filters)
+		query = queryBulkExperiments(query, filters).
+			Where("NOT e.archived").
```

Comment:
so archive is now almost idempotent.. where before you got 
```
ExperimentActionResult{Error: status.Errorf(codes.FailedPrecondition, "Experiment is already archived."), ID: check.ID}
```
why would the `filters == nil` not get the same default filtering for `NOT e.archived` too? seems weird to return 'err already archived' in some paths and to auto filter them in others.